driver/ni/scale.h

```diff
@@ -183,10 +189,12 @@ struct ScaleConfig {
                     << name;
             return;
         }
-        if (type == "linear") scale.linear = LinearScale(parser);
-        else if (type == "map") scale.map = MapScale(parser);
-        else if (type == "polynomial") scale.polynomial = PolynomialScale(parser);
-        else if (type == "table") scale.table = TableScale(parser);
+
+        if (type == "linear") new (&scale.linear) LinearScale(parser);
```

Comment:
Not a fan of using raw pointers when we could be using smart pointers

Reply:
Also, why can't these be passed by value? To prevent the vector being copied? Regardless, we should use smart pointers